app/src/main/java/com/km/real_convenience_store/ui/main/MainViewModel.kt

```diff
@@ -0,0 +1,62 @@
+package com.km.real_convenience_store.ui.main
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.km.real_convenience_store.database.AppDatabase
+import com.km.real_convenience_store.dto.local.FavoriteProductEntity
+import com.km.real_convenience_store.model.ProductUiModel
+import com.km.real_convenience_store.network.NetworkModule
+import com.km.real_convenience_store.ui.product_search.toProductUiModel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class MainViewModel(
+    private val database: AppDatabase,
+) : ViewModel() {
+
+    private val _products = MutableLiveData<List<ProductUiModel>>()
+    val products: LiveData<List<ProductUiModel>>
+        get() = _products
+
+    private val _favoriteProducts = MutableLiveData<List<FavoriteProductEntity>>()
+    val favoriteProducts: LiveData<List<FavoriteProductEntity>>
+        get() = _favoriteProducts
+
+    private var saleType: String? = null
+    var currentPage = 1
+
+    fun setSaleType(type: String?) {
+        saleType =
+            if (saleType.equals(type)) {
```

Comment:
```suggestion
            if (saleType == type) {
```

코틀린에서는 == 으로 문자열 비교해도 괜찮습니당
참고: https://codechacha.com/ko/kotlin-string-compare/